src/java.base/share/classes/java/time/format/DateTimeFormatter.java

```diff
@@ -502,7 +503,9 @@
  * {@code LocalDateTime} to form the instant, with any zone ignored.
  * If a {@code ZoneId} was parsed without an offset then the zone will be
  * combined with the {@code LocalDateTime} to form the instant using the rules
- * of {@link ChronoLocalDateTime#atZone(ZoneId)}.
+ * of {@link ChronoLocalDateTime#atZone(ZoneId)}. If the {@code ZoneId} was
+ * parsed from the zone name that does not indicate daylight saving time, then
+ * the standard offset will be used at the local time-line overlap.
```

Comment:
As written, I would read it as being the generic zone name that gets altered, rather than a zone name that explicitly indicates "winter" time. maybe:

"If the {@code ZoneId} was parsed from a zone name that indicates whether daylight saving time in in operation or not, then that fact will be used to select the correct offset at the local time-line overlap."

Reply:
Modified as suggested.